docs/advanced/encryption.md

```diff
@@ -12,9 +12,10 @@ Hookshot supports end-to-bridge encryption via [MSC3202](https://github.com/matr
 
 In order for hookshot to use encryption, it must be configured as follows:
 - The `experimentalEncryption.storagePath` setting must point to a directory that hookshot has permissions to write files into. If running with Docker, this path should be within a volume (for persistency).
+- If hookshot had ever been started with a Sled-based crypto store, set `experimentalEncryption.useLegacySledStore` to `true`. Otherwise, omit that setting (or set it to `false`) to use the SQLite-based crypto store.
```

Comment:
I am wondering if this is breaking for the present day sled users, or if we care given experimental flags. We should also explain we don't have a migration path for Sled-ers.

Reply:
Better yet: Hookshot should check if a crypto file of an unselected format already exists, and error-out if so.

Reply:
On second thought, the rust-sdk doesn't provide a way to check if a crypto store of a particular type already exists (and trying to check manually isn't straightforward, since the rust-sdk doesn't expose the filenames of its crypto stores), so it's not really worth the effort to be so diligent about it & we can just get away with clearly documenting the importance of this config setting.

Reply:
Looks like I overcomplicated this all along: migrating to a new store type is painless; only migrating back to a previously-used store type (after having deleted/moved the original crypto store files) is troublesome.

AIUI the trouble in question is Hookshot crashing on startup if it tries to use a different E2EE device key than it used previously for that device (i.e. login session).  Switching to a never-before-used crypto store type doesn't hit this issue, since changing the store type also gives Hookshot bots a new device entirely, along with a new device key.

As unlikely as this kind of back-and-forth migration should be in practice, I'll document it just in case.